### PR TITLE
feat: Native Speech Recognition with Voice Activity Detection

### PR DESCRIPTION
**Pull request by yoavf**
## Summary

This PR replaces the audio file recording approach with native speech recognition using `expo-speech-recognition`, providing a much smoother and more intuitive voice input experience.

### Key Improvements

- **🎙️ Native Speech Recognition**: Direct speech-to-text using device's built-in capabilities
- **🔇 Voice Activity Detection (VAD)**: Recording automatically stops after 2 seconds of silence
- **⚡ Instant Recording**: Starts immediately when voice screen opens - no button tap needed
- **📝 Real-time Transcription**: See your words appear as you speak
- **🎯 Better UX**: Clean modal overlay with helpful hints for task structure

### Technical Changes

- Replaced `expo-audio` with `expo-speech-recognition` package
- Removed audio file handling, base64 conversion, and transcription API
- Created new `extract-tasks` API endpoint for text-based task extraction
- Fixed schema validation to properly handle nullable fields
- Added comprehensive test coverage for the new API endpoint

### What's Different

**Before**: 
- Tap to start recording → Tap to stop → Upload audio file → Transcribe → Extract tasks

**Now**: 
- Open voice screen → Speak immediately → Auto-stops when done → Extract tasks from text

### Testing

- ✅ All existing tests pass (103 tests)
- ✅ New tests added for `extract-tasks` API endpoint
- ✅ Tested on Android device with voice input
- ✅ Handles edge cases (no speech, no tasks, multiple tasks)

### Demo

The voice input now shows:
- Pulsing red microphone indicator
- Live transcription display
- Visual hints showing task structure:
  - 📝 What: "Clean the kitchen"
  - 📍 Where: "...in the garage"
  - 📅 When: "...tomorrow at 3pm"

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: app/voice.tsx [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+    const timer = setTimeout(() => {
+      stopListening();
+    }, SILENCE_TIMEOUT_MS);
+    
+    setSilenceTimer(timer);
+  });
+
+  // Handle errors
+  useSpeechRecognitionEvent('error', (event) => {
+    console.error('Speech recognition error:', event.error);
```
Comment (yoavf) [inline app/voice.tsx:105]: replace with logger class
